docker/cli
# Add functionality for `docker kill --all`

**- What I did**
Added `--all` to `docker kill`. This allows a user to easily send a signal to all running containers. 

**- How I did it**
I had made a new func `RequiresArgOrAllFlag` which makes sure a cmd has either args passed to it or has the `--all` flag set. This will be especially useful should `stop` and `rm` get an `--all` flag. Other than that I used the functionality of `ps` to get the containers using the `status=running` filter.

**- How to verify it**
I added new unit test file to cover the `kill --all` functionality as well as the already existing `kill` functionality.

functionality looks like this:
```bash
$ for i in $(seq 3); do docker run -itd alpine sh; done;
b776410ee32895527d804ec7ee08ca7d3ef20c9a4fefc1ebe1384cbfdf8bd051
56245ce81708d8d3eae8c7bc920f353deb2bb6b62d2fa8668b3ebc8ba2c6581e
bd8a998578173134eed2266d51572f861bea3f33aa79b2292a967052329c8b7e
$ docker kill --all
b776410ee32895527d804ec7ee08ca7d3ef20c9a4fefc1ebe1384cbfdf8bd051
56245ce81708d8d3eae8c7bc920f353deb2bb6b62d2fa8668b3ebc8ba2c6581e
bd8a998578173134eed2266d51572f861bea3f33aa79b2292a967052329c8b7e
$ docker kill --all -s SIGINT
no containers running to send SIGINT signal to
$ docker kill --all some-container
"docker kill" requires at least 1 argument or --all flag.
See 'docker kill --help'.

Usage:  docker kill [OPTIONS] CONTAINER [CONTAINER...]

Kill one or more running containers
```

**- Description for the changelog**

Introduce  `kill --all` to send a signal to all running containers

Fixes https://github.com/docker/cli/issues/3565

***Additional Information***
I decided not to make a `rm --all` like the issue conversation suggests because I think there might be a functionality conflict. `docker container prune` requires either a confirmation or a `--force` flag. I think that making `docker rm --all` remove all containers without confirmation could lead to an unpredictable experience for the end user. I'd be open to suggestions, but I'm not sure if it is necessary given `docker prune` exists. Maybe I'm missing something; I would be happy to hear your thoughts about this and about `docker stop --all` :smiley: 


**- A picture of a cute animal (not mandatory but encouraged)**
![image](https://user-images.githubusercontent.com/73488828/166138804-05caa650-7231-4108-8b00-df4ffe6df079.png)
